server/app/views/applicant/EditTiClientView.java

```diff
@@ -90,101 +99,115 @@ private ATag renderBackLink() {
     return link.asAnchorText();
   }
 
-  private DivTag renderEditClientForm(
-      AccountModel account,
+  private DivTag renderAddOrEditClientForm(
+      TrustedIntermediaryGroupModel tiGroup,
+      Optional<AccountModel> mayBeAccount,
       Http.Request request,
-      Optional<Form<EditTiClientInfoForm>> form,
+      Optional<Form<TiClientInfoForm>> form,
       Messages messages) {
-    ApplicantData applicantData = account.newestApplicant().get().getApplicantData();
-    FormTag formTag =
-        form()
-            .withId("edit-ti")
-            .withMethod("POST")
-            .withAction(routes.TrustedIntermediaryController.updateClientInfo(account.id).url());
+    Optional<ApplicantData> mayBeApplicantData = Optional.empty();
+    FormTag formTag;
+    if (mayBeAccount.isPresent()) {
+      mayBeApplicantData =
+          Optional.of(mayBeAccount.get().newestApplicant().get().getApplicantData());
+
+      formTag =
+          form()
+              .withId("edit-ti")
+              .withMethod("POST")
+              .withAction(
+                  routes.TrustedIntermediaryController.editClient(mayBeAccount.get().id).url());
+    } else {
+      formTag =
+          form()
+              .withId("add-ti")
+              .withMethod("POST")
+              .withAction(routes.TrustedIntermediaryController.addClient(tiGroup.id).url());
+    }
     FieldWithLabel firstNameField =
         setStateIfPresent(
             FieldWithLabel.input()
-                .setId("edit-first-name-input")
+                .setId("first-name-input")
                 .setFieldName("firstName")
-                .setLabelText("First name")
+                .setLabelText(messages.at(MessageKey.NAME_LABEL_FIRST.getKeyName()))
                 .setRequired(true)
-                .setValue(applicantData.getApplicantFirstName()),
+                .setValue(setDefaultFirstName(mayBeApplicantData)),
             form,
             TrustedIntermediaryService.FORM_FIELD_NAME_FIRST_NAME,
             messages);
 
     FieldWithLabel middleNameField =
         setStateIfPresent(
             FieldWithLabel.input()
-                .setId("edit-middle-name-input")
+                .setId("middle-name-input")
                 .setFieldName("middleName")
-                .setLabelText("Middle name")
-                .setValue(applicantData.getApplicantMiddleName()),
+                .setLabelText(
+                    messages.at(MessageKey.NAME_LABEL_MIDDLE.getKeyName())
+                        + " "
+                        + messages.at(MessageKey.CONTENT_OPTIONAL.getKeyName()))
```

Comment:
question (non-blocking): It's generally discouraged to concatenate strings together because different languages may have words in different orders. But, it also doesn't seem worthwhile to have to create a new string for "Middle name (optional)". So, this is probably fine for now?

It looks like we also want to add this (optional) indicator for applicants filling out optional questions: https://github.com/civiform/civiform/issues/5510. I'm not sure how we're going to handle internationalization there either. Are we doing to define a message that's `"{0} (optional)"` so that we can plug in any question text into the `{0}` placeholder?

@melanie-exygy do you have any thoughts here?

Reply:
@caitlinshk I think concatenation is fine here because "(optional)" is a separate phrase or indicator, so even in another language, the placement should be the same.  For the applicant optional questions, using `{0} (optional)` would be fine, although I don't see the advantage of that over concatenating, honestly.

Reply:
Sounds good!